.github/workflows/integration-test-fast.yml

```diff
@@ -74,11 +75,11 @@ on:
 jobs:
   integration-tests-fast:
     name: integration-tests-fast
-    runs-on: ${{ inputs.os }}
+    runs-on: ubuntu-latest
```

Comment:
The fact that you removed the configuration here can cause some problems. Since it is set now this job will always run on `ubuntu-latest`. While this is fine, people can set the input to `macos-13` and completely mess up the if conditions down below. I would recommend perhaps removing the `os` from the inputs and adjusting the conditions to use `runner.os` instead of `inputs.os`.